DEFEDIT-1037 Building when a build is already in progress produces error

erikangelin-king opened the pull request:
### User facing changes
* Target menu contains pseudo-target for launching new local dmengine. When launched, details on the instance appears instead. Only one editor-launched instance allowed.
* User can start several dmengine instances from command line and assign them different service ports (DM_SERVICE_PORT) and use them as targets from the editor when they are found via SSDP.
* Can provide a port when manually specifying a target
* More apparent when the editor is actually building vs launching a dmengine

### Technical changes
* dmengine engine port controllable through env var DM_SERVICE_PORT
* dmengine outputs engine port (and log port) using dmLogInfo
* Starting SSDP is no longer a prerequisite for dmengine to launch engine service
* SSDP UDN contains engine port to make it more unique
* editor 2 parses output for engine port

erikangelin-king committed:
DEFEDIT-1037 Further fixes

- Engine outputs port info using dmLogInfo
- Make service port part of ssdp UDN so multiple
  instances can be distinguished when running on same machine
- Engine service port output optional
- Prevent hot reload of no service port found for local dmengine
- Failure to start SSDP service doesn't prevent starting engine service
- Report build/launch progress to both statusbar and console
- Rudimentary error reporting

erikangelin-king committed:
Revert "Revert "Merge pull request #1688 from defold/DEFEDIT-1037""

This reverts commit 0dde28ca92551934cf34137e33d029cce19b2447.

erikangelin-king committed:
Merge branch 'DEFEDIT-1037' into DEFEDIT-1037-dev-merge-1037

erikangelin-king committed:
DEFEDIT-1037 Fixes after revert-revert

- Added console/show!

ragnard-king reviewed (commented):
Re-tested on macOS, seems to work brilliantly! A few comments, but nothing major. I'd like to understand the `Thread/sleep` in `ignore-all-output` though.

erikangelin-king committed:
DEFEDIT-1037 Review fixes

- Removed Thread/sleep
- use :id instead of :process
- proper opts map use
- transducer for list flattening/distinct

erikangelin-king committed:
DEFEDIT-1037 Further review fixes

- Comments in engine code for parsed output
- dmLogError if too many log connections opened
- Broken parsing of log port for launched target
- :process -> :id
